src/unit_tests/ctx/add_to_library.rs

```diff
@@ -0,0 +1,297 @@
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().len(),
+        1,
+        "One request has been sent"
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().get(0).unwrap().to_owned(),
+        Request {
+            url: "https://api.strem.io/api/datastorePut".to_owned(),
+            method: "POST".to_owned(),
+            body: "{\"authKey\":\"auth_key\",\"collection\":\"libraryItem\",\"changes\":[{\"_id\":\"id\",\"removed\":false,\"temp\":false,\"_ctime\":\"2020-01-01T00:00:00Z\",\"_mtime\":\"2020-01-01T00:00:00Z\",\"state\":{\"lastWatched\":null,\"timeWatched\":0,\"timeOffset\":0,\"overallTimeWatched\":0,\"timesWatched\":0,\"flaggedWatched\":0,\"duration\":0,\"video_id\":null,\"watched\":null,\"lastVidReleased\":null,\"noNotif\":false},\"name\":\"name\",\"type\":\"type_name\",\"poster\":null,\"behaviorHints\":{\"defaultVideoId\":null}}]}"
+                .to_owned(),
+            ..Default::default()
+        },
+        "datastorePut request has been sent"
+    );
+}
+
+#[test]
+fn actionctx_addtolibrary_already_added() {
+    #[derive(Model, Debug, Default)]
+    struct Model {
+        ctx: Ctx<Env>,
+    }
+    let meta_preview = MetaPreview {
+        id: "id".to_owned(),
+        type_name: "type_name".to_owned(),
+        name: "name".to_owned(),
+        poster: None,
+        logo: None,
+        description: None,
+        release_info: None,
+        runtime: None,
+        released: None,
+        poster_shape: Default::default(),
+        trailer: None,
+        behavior_hints: Default::default(),
+    };
+    let lib_item = LibItem {
+        id: "id".to_owned(),
+        removed: false,
+        temp: false,
+        ctime: Some(Env::now()),
+        mtime: Env::now(),
+        state: Default::default(),
+        name: "name".to_owned(),
+        type_name: "type_name".to_owned(),
+        poster: None,
+        poster_shape: Default::default(),
+        behavior_hints: Default::default(),
+    };
+    Env::reset();
+    let (runtime, _) = Runtime::<Env, Model>::new(
+        Model {
+            ctx: Ctx {
+                library: LibBucket {
+                    uid: Some("id".to_owned()),
+                    items: vec![("id".to_owned(), lib_item.to_owned())]
+                        .into_iter()
+                        .collect(),
+                },
+                ..Default::default()
+            },
+        },
+        1000,
+    );
+    run(
+        runtime.dispatch(&Msg::Action(Action::Ctx(ActionCtx::AddToLibrary(
+            meta_preview.to_owned(),
+        )))),
+    );
+    assert_eq!(
+        runtime.app.read().unwrap().ctx.library.items.len(),
+        1,
+        "There is one library item in memory"
+    );
+    assert_eq!(
+        runtime
+            .app
+            .read()
+            .unwrap()
+            .ctx
+            .library
+            .items
+            .get(&meta_preview.id)
+            .unwrap()
+            .ctime,
+        Some(&lib_item).unwrap().ctime,
```

Comment:
wrapping a reference in Some and then unwrap it is the worst way of cloning something. seems like youve used some of the previous asserts and just made it to work.
could it be just lib_item.ctime